processor/src/coin/bitcoin.rs

```diff
@@ -0,0 +1,376 @@
+pub struct Block([u8; 32], MBlock);
+impl BlockTrait for Block {
+  type Id = [u8; 32];
+  fn id(&self) -> Self::Id {
+    self.0
+  }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct Fee {
+  pub per_weight: u64,
+  pub mask: u64,
+}
+
+impl Fee {
+  pub fn calculate(&self, weight: usize) -> u64 {
+    ((((self.per_weight * u64::try_from(weight).unwrap()) - 1) / self.mask) + 1) * self.mask
+  }
+}
+
+#[derive(Clone, Debug)]
+pub struct Bitcoin {
+  pub(crate) rpc: Rpc,
+  
+}
+impl Bitcoin {
+  pub async fn new(url: String, username: Option<String>, userpass: Option<String>) -> Bitcoin {
+    Bitcoin { rpc: Rpc::new(url, username.unwrap(), userpass.unwrap()).unwrap()}
+  }
+}
+
+#[derive(Clone, Debug)]
+pub struct Output(SpendableOutput);
+impl From<SpendableOutput> for Output {
+  fn from(output: SpendableOutput) -> Output {
+    Output(output)
+  }
+}
+impl OutputTrait for Output {
+  type Id = [u8; 36];
+
+  //TODO: Implement later
+  fn kind(&self) -> OutputType {
+    OutputType::External
+  }
+
+  fn id(&self) -> Self::Id {
+    let serialized_data = self.0.serialize();
+    let ret: [u8; 36] = serialized_data[0..36].try_into().unwrap();
+    ret
+  }
+
+  fn amount(&self) -> u64 {
+    self.0.amount
+  }
+
+  fn serialize(&self) -> Vec<u8> {
+    self.0.serialize()
+  }
+
+  fn read<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+    SpendableOutput::read(reader).map(Output)
+  }
+}
+
+#[derive(Debug)]
+pub struct SignableTransaction {
+  keys: ThresholdKeys<Secp256k1>,
+  transcript: RecommendedTranscript,
+  height: usize,
+  actual: MSignableTransaction,
+}
+
+#[async_trait]
+impl Coin for Bitcoin {
+  type Curve = Secp256k1;
+
+  type Fee = Fee;
+  type Transaction = PartiallySignedTransaction;
+  type Block = Block;
+
+  type Output = Output;
+  type SignableTransaction = SignableTransaction;
+  type TransactionMachine = TransactionMachine;
+
+  type Address = bitcoin::util::address::Address;
+
+  const ID: &'static [u8] = b"Bitcoin";
+  const CONFIRMATIONS: usize = 3;
+
+  // TODO: Get hard numbers and tune
+  const MAX_INPUTS: usize = 128;
+  const MAX_OUTPUTS: usize = 16;
+
+  fn address(&self, key: ProjectivePoint) -> Self::Address {
+    assert!(key.to_encoded_point(true).tag() == Tag::CompressedEvenY, "YKey is odd");
+    let xonly_pubkey =
+      XOnlyPublicKey::from_slice(&key.to_encoded_point(true).x().to_owned().unwrap()).unwrap();
+    let tweaked_pubkey = bitcoin::schnorr::TweakedPublicKey::dangerous_assume_tweaked(xonly_pubkey);
+    Address::p2tr_tweaked(tweaked_pubkey, bitcoin::network::constants::Network::Regtest)
+  }
+
+  //TODO: Implement later
+  fn branch_address(&self, key: ProjectivePoint) -> Self::Address {
+    self.address(key)
+  }
+
+  async fn get_latest_block_number(&self) -> Result<usize, CoinError> {
+    let block_result = self.rpc.get_height().await.map_err(|_| CoinError::ConnectionError);
+    let block_number = match block_result {
+      Ok(val) => Ok(val as usize),
```

Comment:
Please use explicit casts here, if needed. This RPC function should just return usize from the get go...